src/miner.cpp

```diff
@@ -169,8 +169,8 @@ CTransactionRef BlockAssembler::coinbaseTx(const CScript &scriptPubKeyIn, int _n
 
     // Make sure the coinbase is big enough.
     uint64_t nCoinbaseSize = ::GetSerializeSize(tx, SER_NETWORK, PROTOCOL_VERSION);
-    if (nCoinbaseSize < MIN_TX_SIZE && IsNov152018Scheduled() &&
-        IsNov152018Enabled(Params().GetConsensus(), chainActive.Tip()))
+    if (nCoinbaseSize < MIN_TX_SIZE &&
+        (AreWeOnBCHChain() && IsNov152018Activated(Params().GetConsensus(), chainActive.Tip())))
     {
```

Comment:
for readability, AreWeOnBCHChain() could logically include the test for isnov152018acivated()?

Reply:
no strong opinion, I think that we are going to use it also for future forks while moving from date based to height based activation. 

but I think you have a point, thou. In the future we could just  modifying `AreWeOnBCHChain()`.

will change.